src/Microsoft.AspNetCore.Routing/Template/TemplateBinder.cs

```diff
@@ -340,6 +344,33 @@ private TemplatePart GetParameter(string name)
             return null;
         }
 
+        private bool HasValue(string name, RouteValueDictionary ambientValues, RouteValueDictionary values)
+        {
+            if (values != null)
+            {
+                object value;
+                if (values.TryGetValue(name, out value) && 
+                    IsRoutePartNonEmpty(value))
+                {
+                    return true;
+                }
+            }
+
+            if (ambientValues != null 
```

Comment:
I'm going to have to take some time later to re-familiarize myself with this code... why does a `value` have to be non-empty but an `ambient value` or `default` doesn't?